nomad/fsm.go

```diff
@@ -296,10 +296,21 @@ func (n *nomadFSM) applyDeregisterNode(buf []byte, index uint64) interface{} {
 		panic(fmt.Errorf("failed to decode request: %v", err))
 	}
 
-	if err := n.state.DeleteNode(index, req.NodeID); err != nil {
+	// Messages pre 0.9.3 use a single NodeID
+	var nodeIDs []string
+	if len(req.NodeIDs) == 0 {
+		nodeIDs = append(nodeIDs, req.NodeID)
+	} else if req.NodeID != "" {
+		return fmt.Errorf("invalid request: set NodeIDs instead of NodeID")
```

Comment:
I think this message as ambgious, given at this branch, `NodeIDs` is already set.  Probably `both NodeIDs and NodeIDs are set, expected only one to be set` is clearer?